csw-server/src/main/java/org/fao/geonet/kernel/csw/services/getrecords/es/CswFilter2Es.java

```diff
@@ -199,7 +181,7 @@ protected static String escapeLikeLiteral(String text) {
     }
 
     protected static String convertLikePattern(PropertyIsLike filter) {
-        String result = filter.getLiteral();
+        String result = StringEscapeUtils.escapeJson(filter.getLiteral());
```

Comment:
Would be safer to `escapeJson` the result at the end ? because the filter may contain characters for `escapeChar`, `wildcard` and `singleChar` which will not be processed with `escapeJson` so the next lines may fail to do proper replacement.

Reply:
Done in https://github.com/geonetwork/core-geonetwork/pull/7529/commits/6c18b14ee366f8df47b58d2c7605280a9111e7ff